include/rocksdb/db.h

```diff
@@ -100,6 +100,14 @@ struct Range {
   Range(const Slice& s, const Slice& l) : start(s), limit(l) { }
 };
 
+struct RangePtr {
+  const Slice* start;          // Included in the range
+  const Slice* limit;          // Not included in the range
```

Comment:
my understanding of `GetCleanInputsWithinInterval` is that it'll include the upper endpoint -- do you mind double checking?

Reply:
My mistake, I just copy it from `struct Range`. But we are actually intent on supporting exclusive range end in `DeleteFilesInRange/DeleteFilesInRanges`. Do you think it is OK to add that support in this PR, and make the range end exclusive by default in the new API `DeleteFilesInRanges`?

Reply:
Add support in this PR sounds fine. But I think it'd be more consistent to make it inclusive-inclusive, with an option to make the limit exclusive. This PR (#2863) did a good job of adding the option for `DeleteFilesInRange`, but unfortunately we didn't merge it due to an unsolved test failure.

Reply:
Let's delete `// Not included in the range` since it's optional